cpp/tests/io/json_test.cpp

```diff
@@ -2239,4 +2239,56 @@ TEST_F(JsonReaderTest, MixedTypes)
           expected_list);
 }
 
+TEST_F(JsonReaderTest, MapTypes)
+{
+  using cudf::type_id;
+  // Testing function for mixed types in JSON (for spark json reader)
+  auto test_fn = [](std::string_view json_string, bool lines, std::vector<type_id> types) {
+    std::map<std::string, cudf::io::schema_element> dtype_schema{
+      {"foo1", {data_type{type_id::STRING}}},  // list won't be a string
```

Comment:
Can I understand why a list will not be returned as a string? I know that this might not be in the requirements, but if I ask for a nested type to be a string, I really would like for it to be returned as a string no matter what. I'm not 100% sure how difficult that is to pull off though.

Reply:
I restricted `forced string by input schema` to Struct type alone for 2 reasons.
1. Map type is enclosed with `{}`, which is interpreted as struct in nested JSON reader. so, as per requirement, only struct need to be forced as string.
2. It reduces the search space. Right now, I check if a struct path is present in input schema, which reduces some search space. If list also need to be checked, then the path needs to be built for list too, which is **additional work**, which does not help the map type support requirement. 

It's easy to implement; in cpp/src/io/json/json_column.cu:715 , we need add LIST type also in the condition. Would you prefer to add LIST too?

Reply:
I think this is a missed requirement. Sorry about that. In Spark if I ask for a string and it sees a list or a struct, or really just about any type, it is returned as a string in a very similar way to how the mixed type support works. So yes I would love it if we could get this to work for both list and struct types.

Reply:
No worries @revans2 . We could do it as a separate PR to test it extensively.

Reply:
I filed https://github.com/rapidsai/cudf/issues/15278 for this